hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -634,12 +638,15 @@ public AbfsInputStream openFileForRead(final Path path, final FileSystem.Statist
       // Add statistics for InputStream
       return new AbfsInputStream(client, statistics,
               relativePath, contentLength,
-              populateAbfsInputStreamContext(),
+              populateAbfsInputStreamContext(options),
               eTag);
     }
   }
 
-  private AbfsInputStreamContext populateAbfsInputStreamContext() {
+  private AbfsInputStreamContext populateAbfsInputStreamContext(
+      Optional<Configuration> options) {
+    boolean bufferedPreadDisabled = options.isPresent()
```

Comment:
some getOrElse would be easier

Reply:
Its Configuration in Optional and we need fetch the boolean config out of that. 
So if use orElse() we will end up 
Configuration conf = options.orElse();
boolean bufferedPreadDisabled  = conf != null && conf.getBoolean(FS_AZURE_BUFFERED_PREAD_DISABLE, false);
Otherwise
boolean bufferedPreadDisabled  = options.orElseGet(()->{return new Configuration(false);}).getBoolean("key", false);
Not sure why we should end up creating a new Conf object.
I feel like its better to have current way of hasPresent() check.
WDYT?

Reply:
I just had in my head 
```
boolean bufferedPreadDisabled = options.map(c -> c.getBoolean(FS_AZURE_BUFFERED_PREAD_DISABLE, false).orElse(false);
```
See? Functional all the way

Reply:
Oh ya the map will create new Optional<Boolean> and so we can avoid Config object create.  Ya its fine to create temp Boolean instances and looks cleaner. Thanks.. Updated in PR.

Reply:
sometimes that optional works. Sometimes, well, it sucks. This is one of those times where it seems clean